application/app/services/dataverse/dataverse_hub.rb

```diff
@@ -1,21 +1,38 @@
 module Dataverse
-  class HubRegistry
+  class DataverseHub
     include LoggingCommon
-    CACHE_KEY = 'dataverse_hub_installations'
-    CACHE_EXPIRY = 24.hours.freeze
+    DEFAULT_CACHE_EXPIRY = 24.hours.freeze
     HUB_API_URL = 'https://hub.dataverse.org/api/installation'
 
-    def initialize(url: HubRegistry::HUB_API_URL, http_client: Common::HttpClient.new(base_url: HUB_API_URL), cache: Rails.cache)
+    def initialize(
+      url: HUB_API_URL,
+      http_client: Common::HttpClient.new(base_url: HUB_API_URL),
+      expires_in: DEFAULT_CACHE_EXPIRY
+    )
       @url = url
       @http_client = http_client
-      @cache = cache
+      @expires_in = expires_in
+      @installations = []
+      @last_fetched_at = nil
     end
 
     def installations
-      @cache.fetch(CACHE_KEY, expires_in: CACHE_EXPIRY) do
-        log_info('Fetching Dataverse Hub installations...', {url: @url})
-        fetch_installations
+      if cache_expired?
+        log_info('Fetching Dataverse Hub installations...', { url: @url })
+        result = fetch_installations
+        if result.present?
+          @installations = result
+          @last_fetched_at = Time.current
```

Comment:
how do we ensure this value is stored and used in all HTTP requests? Isn't this class loaded on each request?

Reply:
No, that is why we have the DataverseHubRegistry class.
It is basically a global singleton

Reply:
ok